module02/linkedlist/MajorLinkedList.java

```diff
@@ -0,0 +1,131 @@
+package module02.linkedlist;
+
+public class MajorLinkedList<T> implements LinkedList<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MajorLinkedList() {
+        this.head = null;
+        this.size = 0;
+        this.tail = null;
+    }
+
+    @Override
+
+    public void addHead(T element) {
+        Node<T> newNode = new Node<>(element);
+        if (head == null) {
+            head = newNode;
+        } else {
+            Node<T> current = head;
+            while (current.next != null) {
+                current = current.next;
```

Comment:
The current logic of addHead() is actually adding to the tail, not the head. If the intention is to add to the head, the logic should be simpler, and you should update both head and tail correctly when the list is empty.